advanced/workflows/ancient-egyptian-email.mdx

```diff
@@ -0,0 +1,176 @@
+---
+title: 'Building an Ancient Egyptian Email Assistant'
+description: 'Create an automated email response system with an Ancient Egyptian theme using Shinkai Tools'
+---
+
+# Building an Ancient Egyptian Email Assistant
+
+This tutorial guides you through creating an automated email response system using Shinkai Tools. We'll implement an assistant that responds to emails with the personality of an ancient Egyptian pharaoh, demonstrating the power and flexibility of Shinkai's tool system.
+
+## Prerequisites
+
+Before starting this tutorial:
+1. Install and open Shinkai Desktop
+2. Enable experimental features in Settings
+3. Have an email account with IMAP/SMTP access that supports secure connections (SSL/TLS)
+4. Enable two-factor authentication and generate an app-specific password if supported by your provider
+
+## Email Tool Configuration
+
+First, configure the Email Fetcher and Email Sender tools in the Shinkai Tools section. These are native tools that work out of the box - no additional imports required.
+
+### Email Fetcher (IMAP) Configuration
+Configure the following in the Tool Config page:
+- IMAP Server: Your email provider's IMAP server
+- Port: Your provider's IMAP port (typically uses SSL/TLS)
+- Username: Your email address
+- Password: Your email password or app-specific password
+
+### Email Sender (SMTP) Configuration
+Configure in the same Tool Config page:
+- SMTP Server: Your email provider's SMTP server
+- Port: Your provider's SMTP port (typically uses SSL/TLS)
+- Username: Your email address
+- Password: Your email password or app-specific password
+
+Note: Refer to your email provider's documentation for the specific server addresses and port numbers. Many providers also recommend using app-specific passwords for enhanced security.
+
+## Creating the Email Assistant
+
+Navigate to the Tool Editor through the Shinkai Tools section. Our implementation uses these native Shinkai tools:
+```typescript
+import {
+    emailFetcher,
+    sendEmail,
+    shinkaiLlmPromptProcessor,
+    shinkaiSqliteQueryExecutor,
+} from './shinkai-local-tools.ts';
+```
+
+### Core Components
+
+The assistant consists of four main operations:
+
+1. **Reading New Emails**
```

Comment:
PLease focus on an algorithmic desciption of what the code is doing. For example, dont just say Reading New Email and then hecking for Previously Answered Emails, connect the two actions. For example say something like, first we need to retrieve the email from our inbox using email_fetcher, then if we want to respond to it, first we need to check that we haven't already answered that specific email so we need to qeury our answered emails table